### PR TITLE
Add documentation for how to build a Component

### PR DESCRIPTION
**Pull request by mikee47**
I started this in January but perhaps others would like to assist

### CODE REVIEW

**Comment by slaff** [comment]
Looks good to me :)

File: .gitmodules [hunk 1]
```diff
@@ -1,91 +1,140 @@
...
-	path = Sming/Libraries/Adafruit_SSD1306
-	url = https://github.com/adafruit/Adafruit_SSD1306.git
-	ignore = dirty
+
+
+#################################################################################################
+#
+# `Sming/Libraries` - Optional libraries
+#
+# These must be selected using COMPONENT_DEPENDS setting in your component.mk file
```
Comment (slaff) [inline .gitmodules:130]: > `Sming/Libraries` - Optional libraries ...
> These must be selected using COMPONENT_DEPENDS

We should start also deprecating the use of ARDUINO_LIBRARIES then. But this can be part of a separate PR.

**Comment by slaff** [comment]
@mikee47 Is this PR ready for merging?

**Comment by mikee47** [comment]
@slaff Yes, all done for now thanks.
